.gitignore

```diff
@@ -1,2 +1,3 @@
 node_modules
 package-lock.json
+dist/
```

Comment:
> First of all, this is great - thanks for doing this. I think only very small changes stand between the current state and merging this PR.

Cool.

> I do have some higher level feedback / questions:
> 
> * I understand why individual developers might want to use yarn, but I don't see a strong reason to force people to use it. Everyone should have npm and they have essentially the same features and performance these days. Alternatively, yarn could be installed as a dev dependency so people can just run `npx yarn`.

OK, I've ported it back to npm.

> * It would be useful to add some guidance to the README regarding how non-TypeScript users could import only part of the project. Alternatively, it'd be nice if the build process emitted separate files for each source file (in addition to a combined file) so that non-TypeScript users could use parts of the project without needing a TypeScript compiler / bundler.

I will have to think about this.

> * It seems that build files are omitted from this PR; how would you then propose that the demo remain functional, and how would you propose doing releases of this library? (Understanding this is a merge blocker.)

The demo simply runs as before. I changed almost nothing in the demo files. Just try

```
npm run build
npm run serve
```

> * It looks like source maps aren't emitted? That seems like a significant regression, probably a merge blocker. Should be a small change to add that.

Yes, I'll re-enable them.

> * Is the idea with ESM that a consumer will use Rollup or something like it, with their own custom config, to consume the imports, bundle everything up, and eventually minify it? As is, the UMD version is a much smaller file than the ESM version.

Exactly. People who use modules will have their own bundler, like webpack or similar. They will run their own minifier, of course.

Reply:
Re-enabled sourcemaps, now.

Reply:
> The demo simply runs as before. I changed almost nothing in the demo files. Just try
> npm run build
> npm run serve

Since the artifacts produced by `npm run build` are not tracked by git, the demo will not run on GitHub Pages. (Previously the artifacts were tracked in the `build` folder, whereas with this PR they're in the `dist` folder which is in `.gitignore`.) Similarly, I think library releases would require an additional packaging step so that consumers would not have to build this project in order to make use of it as a dependency.